LKE-13741 fix: freeze dependency tree

WilliamGorge opened the pull request:
## What
Freeze the production dependencies & dependency tree of this repo to make sure this code is frozen in stone

## Why
To avoid new vulnerabilities. This code won’t change and should be frozen in stone as those protocols are really legacy

```
→ npm audit --prod
npm warn config production Use `--omit=dev` instead.
found 0 vulnerabilities
```

## How
- start from node-ldapjs `v2.3.3` as used by `linkurious-server`
- First commit:
  - remove git ignore lockfiles (wtf?)
  - pin production dependencies
  - run `npm shrinkwrap`: https://docs.npmjs.com/cli/v8/commands/npm-shrinkwrap
- Second commit: move [archived](https://github.com/pfmooney/node-ldap-filter) `ldap-filter` into the `lib` folder, and re-install

Tested locally, login with ldap auth provider works properly 

## Tests
```
Suites:   ​54 passed​, ​54 of 54 completed​
Asserts:  ​​​2782 passed​, ​of 2782​
```

gmasclet commented:
@WilliamGorge as discussed, we should consider flattening the internal ldapjs sub-dependencies.

WilliamGorge commented:
@gmasclet i updated the PR:
- I started from node-ldapjs v2.3.3 as used by linkurious-server (before it was `v3` which is not compatible with our server). There is no prod vulnerability in that version either. It was discussed with @davidrapin 
- I did what I did before (pin+shrhinkwrap)
- In this version, ldap does not use the internal `@ldap` modules except `ldap-filter` which I moved to `lib`

It is split properly commit by commit.

Do not hesitate if you have a question.

gmasclet commented:
A couple suggestions:

* I recommend adding a `.nvmrc` to make it clear which npm version is expected to build the project (but no need to invest to migrate to the latest  version, just pin the current version).

* Shall we pin the dev-dependencies too? Not as critical as production ones, but that's a cheap move and it more in line with our usual conventions.

* Do we need to add a `package-lock.json`? We usually always have one, but I guess it's not needed here, as the `npm-shrinkwrap.json` fulfills the same purpose.

---

By the way, the `ldapjs` has been deprecated in npm (see [here](https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions) how it works). Due to that, the following warning is displayed when we run `npm ci` on a repo that depends on it:

```
npm warn deprecated ldapjs@2.3.3: This package has been decomissioned. See https://github.com/ldapjs/node-ldapjs/blob/8ffd0bc9c149088a10ec4c1ec6a18450f76ad05d/README.md
```

We should make sure our fork isn't deprecated. I guess the mere fact of uploading our own packages on npm under a different namespace should be enough.

WilliamGorge commented:
> I recommend adding a .nvmrc to make it clear which npm version is expected to build the project (but no need to invest to migrate to the latest version, just pin the current version).

OK

> Shall we pin the dev-dependencies too? Not as critical as production ones, but that's a cheap move and it more in line with our usual conventions.

I've had issues when trying on this repo with v3.x, i'll re-check

> Do we need to add a package-lock.json? We usually always have one, but I guess it's not needed here, as the npm-shrinkwrap.json fulfills the same purpose.

No, shrinkwrap replaces package lock see the doc https://docs.npmjs.com/cli/v8/commands/npm-shrinkwrap

WilliamGorge commented:
first two points are done, see the new commits

gmasclet commented on .eslintignore:
```diff
@@ -2,3 +2,4 @@ node_modules/
 coverage/
 .nyc_output/
 docs/
+lib/ldap-filter
```
**nit**

add EOF line return
